src/runtime_src/core/tools/common/ReportBOStats.cpp

```diff
@@ -0,0 +1,154 @@
+
+// 3rd Party Library - Include Files
+#include <string>
+#include <boost/format.hpp>
+#include <boost/range/as_array.hpp>
+
+namespace xq = xrt_core::query;
+
+void
+ReportBOStats::getPropertyTreeInternal( const xrt_core::device * pDevice,
+                                     boost::property_tree::ptree & pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data,
+  // Then update this method to do so.
+  getPropertyTree20202(pDevice, pt);
+}
+
+void
+ReportBOStats::getPropertyTree20202( const xrt_core::device * pDevice,
+                                  boost::property_tree::ptree & pt) const
+{
+  boost::property_tree::ptree pt2_list;
+
+  try {
+    auto mem_stat_char = xrt_core::device_query<xrt_core::query::memstat>(pDevice);
+    std::vector<std::string> mem_stat;
+
+    //Convert vector of char from query cmd into vector of strings separated by null char separated by null char
+    boost::split(mem_stat, mem_stat_char, boost::is_any_of("\n"));
+    /* Content of mem_stat now will be as below:
+        [UNUSED] bank0@0x004000000000 (16384MB): 0KB 0BOs
+        [IN-USE] bank1@0x005000000000 (16384MB): 0KB 0BOs
+        [UNUSED] bank2@0x006000000000 (16384MB): 0KB 0BOs
+        [UNUSED] bank3@0x007000000000 (16384MB): 0KB 0BOs
+        [UNUSED] PLRAM[0]@0x003000000000 (0MB): 0KB 0BOs
+        [UNUSED] PLRAM[1]@0x003000400000 (0MB): 0KB 0BOs
+        [UNUSED] PLRAM[2]@0x003000600000 (0MB): 0KB 0BOs
+        [IN-USE] HOST[0]@0x002000000000 (16384MB): 0KB 0BOs
+        [== BO Stats Below ==] NA@0x000000000000 (0MB): 0KB 0BOs
+        [Regular] 0KB 0BOs
+        [UserPointer] 0KB 0BOs
+        [P2P] 0KB 0BOs
+        [DeviceOnly] 0KB 0BOs
+        [Imported] 0KB 0BOs
+        [ExecBuf] 0KB 0BOs
+        [CMA] 0KB 0BOs
+    */
+    bool found = false;
+
+    //Skip DDR usage info by looking for line with string "BO Stats Below"
+    //Following that are BO stats. BO type, size in KB and num of BOs fields separated by blank space
+    for (const auto& line: mem_stat) {
+      if (line.empty())
+        continue;
+
+      //Skip DDR usage stats at begining by looking for beloe string
+      if (!found && line.find("BO Stats Below") != std::string::npos) {
+        found = true;
+	continue;
+      } else if (!found)
+	continue;
+
+      //Capture BOStat fields separated by blank space
+      std::vector<std::string> bo_info;
+      boost::split(bo_info, line, boost::is_any_of("\t "));
+      //Expected fileds are BO type, size in KB and Num of BOs
+      if (bo_info.size() != 3)
+        throw std::runtime_error((boost::format("ERROR: Unexpected format in BO Stats. Line: %s") % line).str());
+
+      boost::property_tree::ptree bo_pt;
+      if (bo_info[0].length() < 5 || bo_info[1].length() < 3 || bo_info[2].length() < 4) {
+        bo_pt.put("bufer_type", "Invalid");
+        bo_pt.put("buffer_count", "n/a");
+        bo_pt.put("memory_used", "n/a");
+        bo_pt.put("memory_used_unit", "n/a");

```

Comment:
We don't need this anymore. Please follow the json that Stephen provided in the email

Reply:
These are required for txt output on console else will require division to be performed to display in other units. And also this provides mem info in two formats for users. So gives users more convenience.

Reply:
Redundant data should NOT be placed into the JSON unless absolutely needed.  I don't believe that this is needed.   

If the TXT output needs additional computation needed that is perfectly acceptable.

Reply:
done